### PR TITLE
Support adding and subtracting timedelta

### PR DESCRIPTION
**Pull request by gabrys**
The following was not possible before:

    d = utcdatetime.now()
    print d + timedelta(days=1)
    print d - timedelta(days=1)
    d += timedelta(minutes=15)

With this update it is

### CODE REVIEW

**Comment by paulfurley** [comment]
Hi @gabrys - thanks for this! Great to see some interest in this library :)

It would be great to capture your desired functionality in some tests - it seems like there are a few separate things:

1. `d` + `timedelta(...)`
2. `d` += `timedelta(...)`
3. `bool(utc_dt1 < utc_dt2)`
4. `bool(utc_dt < naive_dt)`

Also could you give me a hint about why you've deleted the `__isub__` method?

Would you like to add these tests or shall I?

Thanks again!
I'm happy

**Comment by gabrys** [comment]
Hi @paulfurley,

`__isub__` is not needed if you have defined `__sub__`. It's just if you want an optimized version of `-=` operator. Otherwise it'll just do `a = a - b` (so `__sub__` and then re-assignment)

**Comment by gabrys** [comment]
It seems the way to do a proper comparison is supplying `__eq__` and `__lt__` and decorating the class with `@total_ordering` from `functools`.

One more question: is there any reason your code mimics `datetime.datetime`'s `__repr__`? I find it the most annoying feature if you print dates it comes back as a tuple. Is the reason to be able to construct "the same" date by evaluating the printed date?

**Comment by paulfurley** [comment]
To answer your questions:

**Why does repr return eg `utcdatetime(2016, 10, 7, 10, 7, 14, 709940)` ?**

I'm following the [convention outlined here](https://docs.python.org/3/reference/datamodel.html#object.__repr__):

>  If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value

> I find it the most annoying feature if you print dates it comes back as a tuple
I'm not sure how you get that, if I print a datetime it looks like this:

```
>>> print(datetime.datetime.now())
2016-10-07 11:09:37.269221
```

**Also is there any reason you didn't make utcdatetime.utcdatetime a subclass of datetime.datetime? I believe the comparison and +/- timedelta operations would work automatically in this case.**

That's true, but there's a lot of *other* stuff that would happen automatically too which is undesirable, for example any method which returns a `datetime` would have to be overridden to convert to a `utcdatetime`. That would require a lot more code, and a lot more test coverage.

I find the `datetime` object complex to work with because of its naive/timezone-aware split. I want the utcdatetime to be very explicitly timezone-aware and make it hard to muddle with a `datetime`.

To put it another way, if `utcdatetime` inherits from `datetime` it's advertising an **is-a** relationship. That would imply that any time one encounters a `utcdatetime` it is reasonable to treat it as a `datetime`, which is not the case.

Thanks again!

**Comment by gabrys** [comment]
For the printing, you're right printing apparently uses the `__str__` which is great. I had a case with `nose` tests being published to JUnit and somewhere along the way from reading the parameters to constructing the test case name in XML the `__repr__` is involved, so the tuple appears in the XML instead of a nice date representation. But I guess that doesn't mean we should do something about it.

On the note of

> If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value

I'd have the repr return something like:

    utcdatetime('2016-10-07T19:37:12.131Z')

and make `utcdatetime` accept a string like that as a constructor parameter. It's the way I think about dates is always like that and not tuples and to me it's the only logical way of representing a point in time. But I understand this is a personal thing and you might not like that.
